drivers/SmartThings/jbl/src/discovery_mdns.lua

```diff
@@ -0,0 +1,150 @@
+local log = require "log"
+local mdns = require "st.mdns"
+local net_utils = require "st.net_utils"
+
+local discovery_mdns = {}
+
+local function byte_array_to_plain_text(byte_array)
+    local str = ""
+    for _, value in pairs(byte_array) do
+        str = str .. string.char(value)
+    end
+    return str
+end
+
+local function get_text_by_srvname(srvname, discovery_responses)
+    for _,answer_item in pairs(discovery_responses.answers or {}) do
+        if answer_item.kind.TxtRecord ~= nil and answer_item.name == srvname then
+            return answer_item.kind.TxtRecord.text
+        end
+    end
+end
+
+local function get_srvname_by_hostname(hostname, discovery_responses)
+    for _,answer_item in pairs(discovery_responses.answers or {}) do
```

Comment:
nit: we typically use 2 spaces instead of tabs

Reply:
done

Reply:
not done, still looks like 4 spaces

Reply:
change tab from 4 spaces to 2 spaces